### PR TITLE
docs: unify capitalization of headings

### PR DESCRIPTION
**Pull request by marquiz**

### CODE REVIEW

File: docs/get-started/features.md [hunk 1]
```diff
@@ -606,7 +606,7 @@ feature.node.kubernetes.io/override_source-OVERRIDE_VALUE=123
 override.namespace/value=456
 ```
 
-#### A File Example
+#### A File example
```
Comment (zvonkok) [inline docs/get-started/features.md:609]:
```suggestion
#### A file example
```

Comment (marquiz) [reply docs/get-started/features.md]: Thanks, fixed! There was a similar omission on "A Hooks Example" that I also fixed

**Comment by marquiz** [comment]
One more fix to headings pushed...

File: docs/advanced/worker-commandline-reference.md [hunk 2]
```diff
@@ -1,14 +1,14 @@
 ---
-title: "Worker Cmdline Reference"
+title: "Worker cmdline reference"
 layout: default
 sort: 3
 ---
 
-# NFD-Worker Commandline Flags
+# NFD-Worker commandline flags
```
Comment (zvonkok) [inline docs/advanced/worker-commandline-reference.md:7]:
```suggestion
# Nfd-Worker commandline flags
```

Should we treat acronyms differently? Meaning if we use NFD should we have it all capitalized also in a header or do you want to apply the  `Nfd` style?

File: docs/advanced/worker-configuration-reference.md [hunk 3]
```diff
@@ -1,10 +1,10 @@
 ---
-title: "Worker Config Reference"
+title: "Worker config reference"
 layout: default
 sort: 4
 ---
 
-# NFD-Worker Configuration File Reference
+# NFD-Worker configuration file reference
```
Comment (zvonkok) [inline docs/advanced/worker-configuration-reference.md:7]: Here as well...

**Comment by zvonkok** [comment]
I haven't found any guidance on how to treat abbr. in headers but usually it says `do not use it`. If we're using them how should the be written `NFD` or `Nfd` ?

File: docs/advanced/worker-commandline-reference.md (hunk 2 above)
Comment (marquiz) [reply docs/advanced/worker-commandline-reference.md:7]: Oh, these ones. Yes, definitely not *NFD-Worker*, I'd say. Either *NFD-worker* or *Nfd-worker*. In this particular case here we're talking about the name of the binary which is `nfd-worker` so in that sense *Nfd-worker* could make sense. If NFD would be used as a separate abbreviation I'd keep it fullcaps. WDYT @zvonkok? I'm open to suggestions.

**Comment by marquiz** [comment]
> I haven't found any guidance on how to treat abbr. in headers but usually it says `do not use it`.

I think that's prolly a good advise 😄 

> If we're using them how should the abbr. be written `NFD` or `Nfd` ?

As I speculated above, if NFD would be used as a stand-alone abbreviation I'd keep it `NFD`. However as part of the name of the executable I'd prolly use `Nfd`

File: docs/advanced/worker-commandline-reference.md (hunk 2 above)
Comment (zvonkok) [reply docs/advanced/worker-commandline-reference.md:7]: Maybe stupid but what about, 

# The `nfd-worker` commandline flags

Comment (zvonkok) [reply docs/advanced/worker-commandline-reference.md:7]: ### The `nfd-worker` cofinguration file reference

Comment (marquiz) [reply docs/advanced/worker-commandline-reference.md:7]: > The `nfd-worker` commandline flags

Hmm, I'm not a native English-speaker, but, there's something bothering me there 🤔 

How about 
- Command line flags of nfd-master
- Command line flags of nfd-worker
- Configuration reference of nfd-worker
??

Comment (zvonkok) [reply docs/advanced/worker-commandline-reference.md:7]: Yep lets do that, makes more sense, as I said maybe but its definitely `stupid`.  :zany_face:

File: docs/advanced/worker-configuration-reference.md (hunk 3 above)
Comment (marquiz) [reply docs/advanced/worker-configuration-reference.md]: Changed to `Configuration file reference of nfd-worker`

File: docs/advanced/worker-commandline-reference.md (hunk 2 above)
Comment (marquiz) [reply docs/advanced/worker-commandline-reference.md]: Changed to `Commandline flags of nfd-worker`

File: docs/get-started/deployment-and-usage.md [hunk 4]
```diff
@@ -244,7 +244,7 @@ the default interval is set to 60s which is also the default when no
 `-sleep-interval` is specified. Also, the configuration file is re-read on
 each iteration providing a simple mechanism of run-time reconfiguration.
 
-### TLS authentication
+### Tls authentication
```
Comment (zvonkok) [inline docs/get-started/deployment-and-usage.md:247]: Communication security with TLS ?

**Comment by marquiz** [comment]
Changed headings
- `NFD-Master Commandline Flags` -> `Command line flags of nfd-master`
- `NFD-Worker Commandline Flags` -> `Command line flags of nfd-worker`
- `NFD-Worker Configuration File Reference` -> `Configuration reference of nfd-worker`
- `End-to-End Test Configuration File Reference`-> `Configuration file reference of end-to-end tests`

**Comment by zvonkok** [comment]
Ok enough lamenting, only one nitpick and I would say looking good.

File: docs/get-started/deployment-and-usage.md (hunk 4 above)
Comment (marquiz) [reply docs/get-started/deployment-and-usage.md]: OK, changed. Even if this patch is supposed to be just about capitalization 😄

**Comment by zvonkok** [comment]
/lgtm 

Never used PR reviews as pair programming tool :)

**Comment by marquiz** [comment]
> Never used PR reviews as a pair programming tool :)

Thanks. Not the fastest one for that but seemed to work 😊

**Pull request merged**